### ISSUE TITLE
Provide information about specifications from the landing page

### ISSUE THREAD

**Issue by pradyunsg** [pypa/pypa.io#41]
We should in some form link to this page from the landing page on https://pypa.io/ and how it is related to PyPA.

/cc @theacodes @ncoghlan since they likely have the most context here.

**Comment by ncoghlan** [comment]
Agreed we should have a clear link. There's the pypa.io spec page, which describes the process, and the packaging.python.org one that has the specs themselves.

**Comment by pradyunsg** [comment]
We also don't have a link in the pypa.io page that goes to the packaging.python.org section. We should add that too.
